### PR TITLE
nanod | nixos/nanod: init at 20.0

### PR DESCRIPTION
**Pull request by qasaur**
###### Motivation for this change

This change introduces `nanod`, which is the daemon that is used to participate on the nano cryptocurrency network. Much of the derivation is based off the `nano-wallet` package that already exists in `nixpkgs`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/nanod.nix [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        node = { enable-voting = true; };
+        websocket = {
+          address = "::1";
+          enable = true;
+          port = 7078;
+        };
+      };
+    };
+
+    rpcConfig = mkOption {
```
Comment (aanderse) [inline nixos/modules/services/networking/nanod.nix]: Might be more clear to name as: `rpcConfig` -> `rpcConfigFile` and `rpcConfigOptions` -> `rpcConfig`... but just my opinion, take it or leave it :man_shrugging:

File: pkgs/applications/blockchains/nanod/CMakeLists.txt.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index b43f02f6..4470abbf 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -119,7 +119,7 @@ endif (RAIBLOCKS_SECURE_RPC)
+ 
+ include_directories (${CMAKE_SOURCE_DIR})
+ 
+-set(Boost_USE_STATIC_LIBS        ON)
++add_definitions(-DBOOST_LOG_DYN_LINK)
```
Comment (aanderse) [inline pkgs/applications/blockchains/nanod/CMakeLists.txt.patch]: My `cmake`-fu is a bit rusty, but can't we achieve these results from various `cmake` flags in `nix`? :thinking:

File: pkgs/applications/blockchains/nanod/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    fetchSubmodules = true;
+  };
+
+  # Use a patch to force dynamic linking
+  patches = [
+    ./CMakeLists.txt.patch
+  ];
+
+  cmakeFlags = let
+    options = {
```
Comment (aanderse) [inline pkgs/applications/blockchains/nanod/default.nix]: This abstraction seems a bit overkill, especially for two simple values.

**Review by aanderse — CHANGES_REQUESTED** [review]
Thanks for your contribution to nixpkgs! I'm glad you want to join the maintainer team. I've left some comments which I hope are helpful.

I hope I can convince @mmahut to take a quick look at this too. I'm pretty sure crypto stuff is his cup of tea, and I'm out of my element here.

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**